backends/guest/common/read.c

```diff
@@ -124,14 +124,17 @@ int print_variables(const uint8_t *buffer, size_t buffer_size, const uint8_t *va
 {
 	int rc;
 	ssize_t esl_data_size = buffer_size, cert_size;
```

Comment:
agreed, we should split this into smaller functions.

But if you are going to reduce scope of variables then you might as well go all the way:
 - `data_size` is useless and can be removed and replaced with `cert_size`
 - `cert_size`, `esd` can be declared in the second loop
 - `signature_type`, `sig_list` can go in first loop
 -  I don't think `offset` is being used properly in this function at all, in the first loop we use it for offset to the next ESL header, in the second, loop we use it as an offset to the ESL data. I doubt this works if there are ever two appended ESL structures (which I believe is possible with dbx, unless something changed)

Reply:
As this was mostly to silence cppcheck warnings, most of the rest of the scope reductions will probably be removed with a function rewrite. Skipping for now.